tests/main/snapd-reexec/task.yaml

```
@@ -77,7 +77,19 @@ execute: |
     fi
 
     echo "Ensure we re-exec by default"
-    /usr/bin/env SNAPD_DEBUG=1 snap list 2>&1 | MATCH "DEBUG: restarting into \"$SNAPD_MOUNT_DIR/current/usr/bin/snap\""
+    IS_REEXEC=true
+    if [ "$SNAPD_SRC" = core ] && tests.info is-snapd-pkg-repo; then
+        # In this scenario it is possible that the core snap version is older
+        # than the distribution package because the core snap comes from latest/edge
+        # which is not updated regularly.
+        # TODO: Compare manually the snap and deb versions and determine which string needs to be in MATCH
+        if /usr/bin/env SNAPD_DEBUG=1 snap list 2>&1 | MATCH "snap \(at .*\) is older \(.*\) than distribution package"; then
+            IS_REEXEC=false
+        fi
+    fi
+    if [ "$IS_REEXEC" = true ]; then
+        /usr/bin/env SNAPD_DEBUG=1 snap list 2>&1 | MATCH "DEBUG: restarting into \"$SNAPD_MOUNT_DIR/current/usr/bin/snap\""        
```

this would work too, wouldn't it?
```suggestion
        SNAPD_DEBUG=1 snap list 2>&1 | MATCH "DEBUG: restarting into \"$SNAPD_MOUNT_DIR/current/usr/bin/snap\""        
```